### DISCUSSION TITLE
Not getting the event "session_proposal" in Android

### DISCUSSION THREAD

**Discussion by longndmta** [WalletConnect/walletconnect-monorepo#3026] [Developer Support]
Excuse me! I use wallet connect v2 with react native app. iOS works fine. Android not working. It seems that i am not getting the event "session_proposal". Do I need any more config in Android?

**Comment by crypblizz8** [comment]
Hey there, don't believe you need extra config. But can you share of your code for the config?

@ignaciosantise can also help look into this

**Comment by longndmta** [comment]
I followed steps in the tutorial.
iOS works fine. Android not working
```
const PROJECT_ID = '3f7b74fc7300c51d6e56dedb695eeefb';
const core = new Core({
        projectId: PROJECT_ID,
      });
const web3wallet = await Web3Wallet.init({
    core, // <- pass the shared `core` instance
    metadata: {
      name: 'Demo React Native Wallet',
      description: 'Demo RN Wallet to interface with Dapps',
      url: 'www.walletconnect.com',
      icons: []
    }
  });
web3wallet.on("session_proposal", async (proposal) => {
    const { id, params } = proposal;
    const {
      proposer,
     } = params;
    console.log(`WC2::session_proposal id=${id}`, params);
});
const paired = await web3Wallet.core.pairing.pair({
          uri: cleanUri,
        });
```

**Reply by ignaciosantise** [reply]
hey @longndmta as i can see, that code should work good. Can you share an example repository with the reproducible issue? So i can test it out by myself.

**Comment by longndmta** [comment]
Yes it works fine with IOS like i said. However not with Android.
So i want to ask do i need any more native configuration for Android?

**Reply by ignaciosantise** [reply]
No, there's no extra config for android. Can you share an example repository with the reproducible issue? So i can test it out by myself.

**Comment by kale1d** [comment]
I'm having a similar issue when building the apk. When is running with metro is working just fine but when building the apk web3walletclient is returning undefined. Any suggestion ? Thx!

**Reply by ignaciosantise** [reply]
@kale1d were you able to solve this? Can you create a public example where i can reproduce the issue?
I've created several APKs with web3wallet and i didn't have issues

**Comment by fullstackYao** [comment]
@longndmta I have also encountered this problem. Have you resolved it
